conda_pip/main.py

```diff
@@ -110,13 +125,71 @@ def run_pip_install(
         command.append("-" + ("v" * verbosity))
     if force_reinstall:
         command.append("--force-reinstall")
-    if yes:
-        command.append("--yes")
     if upgrade:
         command.append("--upgrade")
     command.extend(specs)
 
     logger.info("pip install command: %s", command)
     process = run(command)
-    if process.returncode:
-        return process.returncode
+    return process.returncode
+
+
+def place_externally_managed(prefix: os.PathLike = None) -> Path:
+    """
+    conda-pip places its own EXTERNALLY-MANAGED file when it is installed in an environment.
+    We also need to place it in _new_ environments created by conda. We do this by implementing
+    some extra plugin hooks.
+    """
+    base_dir = get_env_stdlib(prefix or sys.prefix)
+    externally_managed = Path(base_dir, "EXTERNALLY-MANAGED")
+    if externally_managed.exists():
+        return
+    logger.info("Placing EXTERNALLY-MANAGED in %s", base_dir)
+    shutil.copy(HERE / "data" / "EXTERNALLY-MANAGED", externally_managed)
+    return externally_managed
+
+
+def ensure_target_env_has_externally_managed(command: str):
+    """
+    post-command hook to ensure that the target env has the EXTERNALLY-MANAGED file
+    even when it is created by conda, not 'conda-pip'.
+    """
+    if os.environ.get("CONDA_BUILD_STATE") == "BUILD":
+        return
+    base_prefix = Path(context.conda_prefix)
+    target_prefix = Path(context.target_prefix)
+    if base_prefix == target_prefix:
+        return
+    # ensure conda-pip was explicitly installed in base env (and not as a dependency)
+    requested_specs_map = History(base_prefix).get_requested_specs_map()
+    if requested_specs_map and "conda-pip" not in requested_specs_map:
+        return
+    prefix_data = PrefixData(target_prefix)
+    if command in {"create", "install", "update"}:
+        # ensure target env has pip installed
+        if not list(prefix_data.query("pip")):
+            return
+        # Check if there are some leftover EXTERNALLY-MANAGED files from other Python versions
+        if command != "create" and os.name != "nt":
+            for python_dir in Path(target_prefix, "lib").glob("python*"):
+                if python_dir.is_dir():
+                    externally_managed = Path(python_dir, "EXTERNALLY-MANAGED")
+                    if externally_managed.exists():
+                        externally_managed.unlink()
+        place_externally_managed(target_prefix)
+    else:  # remove
+        if list(prefix_data.query("pip")):
```

Comment:
Any exception expected with this call?

Reply:
`.query()` returns a generator. Worst case it's empty because there were no matches.